ooni/nettests/blocking/web_connectivity.py

```diff
@@ -0,0 +1,546 @@
+
+        if control_body_length == experiment_body_length:
+            rel = float(1)
+        elif control_body_length == 0 or experiment_body_length == 0:
+            rel = float(0)
+        else:
+            rel = float(control_body_length) / float(experiment_body_length)
+
+        if rel > 1:
+            rel = 1/rel
+
+        self.report['body_proportion'] = rel
+        if rel > float(self.factor):
+            return True
+        else:
+            return False
+
+    def compare_titles(self, experiment_http_response):
+        experiment_title = extractTitle(experiment_http_response.body).strip()
+        control_title = self.control['http_request']['title'].strip()
+        first_exp_word = experiment_title.split(' ')[0]
+        first_ctrl_word = control_title.split(' ')[0]
+        if len(first_exp_word) < 5:
+            # We don't consider to match words that are shorter than 5
+            # characters (5 is the average word length for english)
+            return False
+        return (first_ctrl_word.lower() == first_exp_word.lower())
+
+    def compare_http_experiments(self, experiment_http_response):
+
+        self.report['body_length_match'] = \
+            self.compare_body_lengths(experiment_http_response)
+
+        self.report['headers_match'] = \
+            self.compare_headers(experiment_http_response)
+
+        if str(self.control['http_request']['status_code'])[0] != '5':
+            self.report['status_code_match'] =  (
+                self.control['http_request']['status_code'] ==
+                experiment_http_response.code
+            )
+
+        self.report['title_match'] = self.compare_titles(experiment_http_response)
+
+    def compare_dns_experiments(self, experiment_dns_answers):
+        if self.control['dns']['failure'] is not None and \
+                self.control['dns']['failure'] == self.report['dns_experiment_failure']:
+            self.report['dns_consistency'] = 'consistent'
+            return True
+
+        control_addrs = set(self.control['dns']['addrs'])
+        experiment_addrs = set(experiment_dns_answers)
+
+        if control_addrs == experiment_addrs:
+            return True
+
+        for experiment_addr in experiment_addrs:
+            if is_public_ipv4_address(experiment_addr) is False:
+                return False
+
+        if len(control_addrs.intersection(experiment_addrs)) > 0:
+            return True
+
+        experiment_asns = set(map(lambda x: geoip.IPToLocation(x)['asn'],
+                              experiment_addrs))
+        control_asns = set(map(lambda x: geoip.IPToLocation(x)['asn'],
+                           control_addrs))
+
+        # Remove the instance of AS0 when we fail to find the ASN
+        control_asns.discard('AS0')
+        experiment_asns.discard('AS0')
+
+        if len(control_asns.intersection(experiment_asns)) > 0:
+            return True
+
+        return False
```

Comment:
I think for this I would like to leave it explicit like so, as to make it clearer what are the all the possible outcomes of this comparison.